design/experimental-new-features-in-modular.md

```diff
@@ -0,0 +1,204 @@
+# Shipping experimental new features in Modular
+
+## Background
+
+As the Azure SDK for JavaScript continues to evolve, it is important for us to have a dedicated space to ship experimental features within our own client or language. By providing a separate subpath export for experimental features alongside with the regular release of Modular, we can gather valuable feedback from our customers who have tried these new features. This approach enables us to receive early insights and iterate on the experimental features while ensuring that our mainline releases remain stable and reliable.
+
+## Questions
+
+1. **Version picking strategy**  
+  One latest GA + One latest preview version.  
+  If the latest preview version < latest GA version, then we will use latest GA version as the experiment version.  
+  _Questions_:
+    1. do we allow customers to set their own preview version?
+
+1. **Feature picking strategy**  
+  Let's say we have Feature 1, 2, 3, 4 to experiment
+  _Questions_:
+    1. Is it possible that we want features 1, 2 in one RP but features 3, 4 in another RP?
+    1. If service team want a pure preview with their service version without any client side preview features, is this allowed?  
+
+1. **Feature staging status change**  
+  _Questions_:
+   1. Is it possible that a feature could be GA or withdrew based on customers feedback?
```

Comment:
I think whether or not a feature as I understand it is stable or experimental is determined by the service itself and not by the SDK. We just do our best to represent the stability of various API capabilities in the service. They may choose to GA a feature but would be subject to Azure rules about breaking changes with respect to deprecating and removing a feature.